cmd/kro/commands/publish/publish.go

```diff
@@ -0,0 +1,235 @@
+	Long: "The publish command takes an OCI image tarball, created by the 'package' command, " +
+		"and pushes it to a specified container registry.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if publishConfig.ociTarballPath == "" {
+			return fmt.Errorf("path to the OCI tarball is required, please use the --file flag")
+		}
+
+		if publishConfig.remoteRef == "" {
+			return fmt.Errorf("remote reference is required, please use the --ref flag")
+		}
+
+		if publishConfig.username == "" {
+			fmt.Print("Username: ")
+			reader := bufio.NewReader(os.Stdin)
+			username, err := reader.ReadString('\n')
+			if err != nil {
+				return fmt.Errorf("failed to read username: %w", err)
+			}
+			publishConfig.username = strings.TrimSpace(username)
+		}
+
+		if publishConfig.password == "" {
+			fmt.Print("Password: ")
+			bytePassword, err := term.ReadPassword(int(os.Stdin.Fd()))
+			fmt.Println()
+			if err != nil {
+				return fmt.Errorf("failed to read password: %w", err)
+			}
+			publishConfig.password = string(bytePassword)
+		}
+
+		ctx := context.Background()
+		repo, err := remote.NewRepository(publishConfig.remoteRef)
+		if err != nil {
+			return fmt.Errorf("failed to create remote repository for %s: %w", publishConfig.remoteRef, err)
+		}
+
+		credentialFunc := func(ctx context.Context, registry string) (auth.Credential, error) {
+			return auth.Credential{
+				Username: publishConfig.username,
+				Password: publishConfig.password,
+			}, nil
+		}
+
+		repo.Client = &auth.Client{
+			Client:     retry.DefaultClient,
+			Cache:      auth.NewCache(),
+			Credential: credentialFunc,
+		}
+
+		if publishConfig.username == "" || publishConfig.password == "" {
+			return fmt.Errorf("username and password are required")
+		}
+
+		tempDir, err := os.MkdirTemp("", "kro-publish-*")
+		if err != nil {
+			return fmt.Errorf("failed to create temp dir: %w", err)
+		}
+		defer func() {
+			if err := os.RemoveAll(tempDir); err != nil {
+				fmt.Println("Warning: failed to remove temp dir", tempDir, ":", err)
+			}
+		}()
+
+		fmt.Println("Extracting", publishConfig.ociTarballPath, "to", tempDir, "...")
+		if err := untar(publishConfig.ociTarballPath, tempDir); err != nil {
+			return fmt.Errorf("failed to extract OCI tarball: %w", err)
+		}
+
+		store, err := oci.New(tempDir)
+		if err != nil {
+			return fmt.Errorf("failed to open OCI layout at %s: %w", tempDir, err)
+		}
+
+		rootDesc, err := findRootManifestDescriptor(tempDir)
+		if err != nil {
+			return fmt.Errorf("failed to find root manifest in OCI layout: %w", err)
+		}
+		fmt.Println("Found manifest to push:", rootDesc.Digest)
+
+		fmt.Println("Publishing to", publishConfig.remoteRef, "...")
+		_, err = oras.Copy(ctx, store, rootDesc.Digest.String(), repo, publishConfig.remoteRef, oras.DefaultCopyOptions)
+		if err != nil {
+			return fmt.Errorf("failed to publish OCI image: %w", err)
+		}
+
+		fmt.Println("Successfully published OCI image to", publishConfig.remoteRef)
+		return nil
```

Comment:
I think we're missing the `kro package` command, i wouldn't expect `kro publish` to do any package/extracting here. It should just get a tarball file and sent it to OCI